tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/security/SecurityTest.java

```diff
@@ -1412,12 +1413,26 @@ public void testSendMessageUpdateRoleCached() throws Exception {
 
       securityManager.getConfiguration().addRole("auser", "receiver");
 
-      session.createConsumer(SecurityTest.queueA);
+      Wait.assertTrue(() -> {
+         try {
+            session.createConsumer(SecurityTest.queueA);
+            return true;
+         } catch (Exception e) {
+            return false;
+         }
+      }, 2000, 100);
 
       // Removing the Role... the check should be cached, so the next createConsumer shouldn't fail
       securityManager.getConfiguration().removeRole("auser", "receiver");
```

Comment:
I don't understand this...

Why can't we just use the previous operation on clearing the cache.

I don't think this is a valid test change. The test was written to validate the cache cleared. 


if caching becomes a valid option now, we should rather just remove the test.

Reply:
That test was meant to (among other things) exercise the _authorization_ cache. However, with the introduction of the authentication cache the semantics changed a bit. I've updated the test to make this more clear. Thanks for the review.